tutorials/tts/Fastpitch_Training_GermanTTS.ipynb

```diff
@@ -0,0 +1,1112 @@
+   "source": [
+    "def evaluate_spec_fastpitch_ckpt(spec_gen_model, v_model, test):\n",
+    "    with torch.no_grad():\n",
+    "        torch.manual_seed(seed)\n",
+    "        torch.cuda.manual_seed(seed)\n",
+    "        torch.backends.cudnn.enabled = True\n",
+    "        torch.backends.cudnn.benchmark = False\n",
+    "        parsed = spec_gen_model.parse(str_input=test, normalize=True)\n",
+    "        spectrogram = spec_gen_model.generate_spectrogram(tokens=parsed)\n",
+    "        print(spectrogram.size())\n",
+    "        audio = v_model.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "\n",
+    "    spectrogram = spectrogram.to('cpu').numpy()[0]\n",
+    "    audio = audio.to('cpu').numpy()[0]\n",
+    "    audio = audio / np.abs(audio).max()\n",
+    "    return audio, spectrogram"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7f66e1cd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# load models\n",
+    "vocoder_model = HifiGanModel.from_pretrained(hfg_ngc, strict=False).eval().cuda()\n",
+    "spec_gen_model = FastPitchModel.restore_from(fastpitch_ckpt_path).eval().cuda() if \".nemo\" in fastpitch_ckpt_path else FastPitchModel.load_from_checkpoint(checkpoint_path=fastpitch_ckpt_path).eval().cuda()\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "db40fb75",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "\n",
+    "audio, spectrogram = evaluate_spec_fastpitch_ckpt(spec_gen_model, vocoder_model, test)\n",
+    "\n",
+    "# visualize the spectrogram\n",
+    "if spectrogram is not None:\n",
+    "    imshow(spectrogram, origin=\"lower\")\n",
+    "    plt.show()\n",
+    "\n",
+    "# audio\n",
+    "print(\"original audio\")\n",
+    "ipd.display(ipd.Audio(data_path+test_id+'.wav', rate=22050))\n",
+    "print(\"predicted audio\")\n",
+    "ipd.display(ipd.Audio(audio, rate=22050))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd1c0806",
+   "metadata": {},
+   "source": [
+    "We see that audio quality is not as good as we expect. One of the ways mentioned in the [FastPitch_Finetuning.ipynb](FastPitch_Finetuning.ipynb) tutorial is to finetune HiFi-GAN. Lets try that out next!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d450d10a",
+   "metadata": {},
+   "source": [
+    "# Finetuning HiFi-GAN\n",
+    "\n",
+    "Improving speech quality by Finetuning HiFi-GAN on synthesized mel-spectrograms from Fastpitch."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "77a5b3f2",
+   "metadata": {},
+   "source": [
+    "## Generate synthetic mels\n",
+    "\n",
+    "In order to generate synthetic mels, we take the latest checkpoint from Fastpitch training and predict spectrograms for each of the input records in `train_phonemes_manifest.json`, `test_phonemes_manifest.json` and `val_phonemes_manifest.json`.\n",
+    "\n",
+    "```python\n",
```

Comment:
Please make the following code runnable.

Also, since it is not significantly different from the code in the fine-tuning tutorial, can you add a couple of lines to explicitly say that, and explain what _has_ changed?